sdk/src/transaction_context.rs

```diff
@@ -218,8 +251,34 @@ impl TransactionContext {
         if self.instruction_stack.is_empty() {
             return Err(InstructionError::CallDepth);
         }
+        // Verify (before we pop) that the total sum of all lamports in this instruction did not change
+        let detected_an_unbalanced_instruction = if self
+            .is_early_verification_of_account_modifications_enabled()
+        {
+            self.get_current_instruction_context()
+                .and_then(|instruction_context| {
+                    // Verify all executable accounts have no outstanding refs
+                    for account_index in instruction_context.program_accounts.iter() {
+                        self.get_account_at_index(*account_index)?
```

Comment:
hey, i know you're itching to merge this great pr... but.

this `?` can create unbalanced `self.instruction_stack`...

Reply:
oh, never mind...

 this is actually is in a `.and_then()` closure?